src/Data/Array.js

```diff
@@ -290,7 +290,7 @@ exports.sortByImpl = (function () {
     while (i < mid && j < to) {
       x = xs2[i];
       y = xs2[j];
-      c = fromOrdering(compare(x)(y));
+      c = fromOrdering(compare(x)(y)) || i - j;
```

Comment:
This is equivalent to removing the `else if (c === 0) { /* … */}` later right? I think doing so would be clearer.

Reply:
Simplified here 5524a9fa3d0194523e87e25cf385adbfd3b63a26